test/cfa/sudoers_test.rb

```diff
@@ -0,0 +1,183 @@
+        .with(hash_including(file_path: file_path))
+        .and_call_original
+
+      file = described_class.load(file_path: file_path)
+
+      expect(file.loaded?).to eq(true)
+    end
+  end
+
+  describe "#load" do
+    it "loads the file content" do
+      expect { sudoers.load }.to change { sudoers.loaded? }.from(false).to(true)
+    end
+  end
+
+  describe "#users_specs" do
+    before { sudoers.load }
+
+    context "when the loaded file already contains user spec" do
+      it "returns a collection holding all user specs" do
+        expect(sudoers.users_specs).to eq([
+          { "user" => "ALL", "host" => "ALL", "run_as" => "ALL", "commands" => ["ALL"], },
+          { "user" => "root", "host" => "ALL", "run_as" => "ALL", "commands" => ["ALL"], }
+        ])
+      end
+    end
+
+    context "when the loaded file does not contain user specs yet" do
+      let(:file_path) { File.join(DATA_PATH, "sudoers_no_specs_example") }
+
+      it "return an empty collection" do
+        expect(sudoers.users_specs).to eq([])
+      end
+
+      context "but a new user specs were added" do
+        before do
+          sudoers.add_user(yast_user_spec)
+          sudoers.add_user(cli_user_spec)
+        end
+
+        it "returns a collection holding all user specs" do
+          expect(sudoers.users_specs).to eq([yast_user_spec, cli_user_spec])
+        end
+      end
+    end
+  end
+
+  describe "#clean_users" do
+    before { sudoers.load }
+
+    it "removes all users specifications" do
+      initial_users_specs = sudoers.users_specs
+
+      sudoers.clean_users
+
+      expect(sudoers.users_specs).to_not eq(initial_users_specs)
+      expect(sudoers.users_specs).to eq([])
+    end
+  end
+
+  describe "#save" do
+    let(:file_path) { File.join(tmpdir, "sudoers") }
+    let(:tmpdir) { Dir.mktmpdir }
+
+    before do
+      # Use a copy of the example file
+      FileUtils.cp(example_file, file_path)
+
+      sudoers.load
+    end
+
+    after do
+      FileUtils.remove_entry(tmpdir)
+    end
+
+    context "when the file does not contain user specifications" do
+      let(:example_file) { File.join(DATA_PATH, "sudoers_no_specs_example") }
+
+      context "nor does it include the #includedir sudoers.d directive" do
+        let(:example_file) { File.join(DATA_PATH, "sudoers_old_example") }
+
+        it "writes new specifications at the end of file" do
+          sudoers.add_user(yast_user_spec)
+          sudoers.add_user(cli_user_spec)
+          sudoers.save
+
+          expect(File.read(file_path)).to match(/.*cli ALL = \(ALL\) \/sbin\/yast\Z/)
+        end
+      end
+
+      context "but does include the #includedir sudoers.d directive" do
+        it "writes new specifications before it" do
+          sudoers.add_user(yast_user_spec)
+          sudoers.add_user(cli_user_spec)
+          sudoers.save
+
+          expect(File.read(file_path)).to match(/.*yast.*cli.*\#includedir.*sudoers\.d.*/m)
```

Comment:
I worry that this does not test that it write immediatelly before, but somewhere before.

Reply:
:innocent: I didn't understand your concern, sorry.

Reply:
well, problem is that `.*` can match any number of `\n` not just one. So it will match even
```
.*yast.*cli.
bla
bla
bla
#includedir sudoers.d
```

Reply:
Ah, I see.

In this case, the test still being valid since the intention is that user specifications will be written before the directive. No matter if immediately before somewhere before. 

For sure, we expect a consistent behavior, so I'll improve the test :)

Thanks